cloudify_azure/resources/deployment.py

```diff
@@ -0,0 +1,117 @@
+from azure.mgmt.resource import ResourceManagementClient
+from azure.mgmt.resource.resources.models import DeploymentMode
+
+
+class Deployment(object):
+
+    def __init__(self, logger, credentials, name, timeout=None):
+        self.resource_group = name
+        self.logger = logger
+        self.timeout = timeout
+        self.credentials = ServicePrincipalCredentials(
+            client_id=str(credentials['client_id']),
+            secret=str(credentials['client_secret']),
+            tenant=str(credentials['tenant_id'])
+        )
+        self.client = ResourceManagementClient(
+            self.credentials, str(credentials['subscription_id']))
+
+        self.logger.info("Use subscription: {}"
+                         .format(credentials['subscription_id']))
+
+    def create(self, template, params, location):
+        """Deploy the template to a resource group."""
+        self.logger.info("Client resource...")
+        self.client.resource_groups.create_or_update(
+            self.resource_group,
+            {
+                'location': location
+            }
+        )
+
+        self.logger.info("Create deployment...")
+
+        parameters = {str(k): {'value': str(v)} for k, v in params.items()}
+
+        if isinstance(template, basestring):
+            template = json.loads(template)
+
+        deployment_properties = {
+            'mode': DeploymentMode.incremental,
+            'template': template,
+            'parameters': parameters
+        }
+
+        deployment_async_operation = self.client.deployments.create_or_update(
+            self.resource_group,
+            'azure-sample',
+            deployment_properties
+        )
+        self.logger.info("Wait deployment...Timeout: {}"
+                         .format(repr(self.timeout)))
+        deployment_async_operation.wait(timeout=self.timeout)
+
+    def delete(self):
+        """Destroy the given resource group"""
+        self.logger.info("Delete resource groups...")
+        deployment_async_operation = self.client.resource_groups.delete(
+            self.resource_group
+        )
+        self.logger.info("Wait delete deployment...Timeout: {}"
+                         .format(repr(self.timeout)))
+        deployment_async_operation.wait(timeout=self.timeout)
+
+
+@operation
+def create(ctx, **kwargs):
+    properties = {}
+    properties.update(ctx.node.properties)
+    properties.update(kwargs)
+    ctx.logger.info("Create: {}".format(repr(properties['name'])))
+    deployment = Deployment(ctx.logger, properties['azure_config'],
+                            properties['name'],
+                            timeout=properties.get('timeout'))
+
+    # load template
+    template = properties.get('template')
+    if not template:
+        ctx.logger.info("Will be used {} as template"
+                        .format(repr(properties.get('template_file'))))
+        template = ctx.get_resource(properties['template_file'])
```

Comment:
We should probably handle failure here, like [here](https://github.com/cloudify-incubator/cloudify-kubernetes-provider/blob/master/examples/cluster_blueprint/scripts/cfy_autoscale/install.py#L61), since normally this retries, and it should really raise a NonRecoverableError.